### PR TITLE
Pass registryPullSecrets from Installation to Execution and DeployItems

### LINKED ISSUE #125: Pass RegistryPullSecrets to DeployItems
**How to categorize this issue?**

/area deployers
/kind enhancement
/priority normal

**What would you like to be added**:

Currently it is possible to set `registryPullSecrets` on Installations to specify registry secrets for Component Descriptors and blueprints.
These secrets are only used in the landscaper but are not available for deployers.
As deployers also need access to the registry (helm charts, component descriptors or other resources) these secrets should also be passed to the deployers.

Therefore, we would need to enhance the `DeployItem` CR and add a optional `registryPullSecrets` attribute.

### PR DESCRIPTION
**Pull request by loewenstein**
**How to categorize this PR?**
/area deployers
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR picks up the `registryPullSecrets` from the `Installation`, passes it on to `Execution` and `DeployItem`.
The Helm and Container deployer pick up the `registryPullSecrets` and use them to fetch Helm chart and execution container.

**Which issue(s) this PR fixes**:
Fixes #125 

**Special notes for your reviewer**:
I've provided unit tests where `*_test.go` files where already in place.
I checked both Helm and Container `DeployItems` manually for e2e functionality.

**Release note**:
```feature user
Helm charts and execution containers can be fetched from private OCI registries with the credentials provided as `Installation.spec.registryPullSecrets`
```

### CODE REVIEW

**Comment by gardener-robot-ci-3** [comment]
Thank you @loewenstein for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: pkg/deployer/helm/helm.go [hunk 1]
```diff
@@ -87,6 +90,47 @@ func New(log logr.Logger, helmconfig *helmv1alpha1.Configuration, kubeClient cli
 	}, nil
 }
 
+func createOCIClient(ctx context.Context, log logr.Logger, client client.Client, item *lsv1alpha1.DeployItem, config *helmv1alpha1.Configuration, componentsRegistryMgr *componentsregistry.Manager) (ociclient.Client, error) {
```
Comment (schrodit) [inline pkg/deployer/helm/helm.go:93]: we have the structure that non-exposed helper functions are by below exposed functions in a file.
Can you put the helper functions to the bottom of the file.

File: pkg/deployer/helm/helm.go [hunk 2]
```diff
@@ -87,6 +90,47 @@ func New(log logr.Logger, helmconfig *helmv1alpha1.Configuration, kubeClient cli
...
+		ociclient.WithCache{Cache: componentsRegistryMgr.SharedCache()},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return ociClient, nil
+}
+
+func resolveSecrets(ctx context.Context, client client.Client, secretRefs []lsv1alpha1.ObjectReference) ([]corev1.Secret, error) {
```
Comment (schrodit) [inline pkg/deployer/helm/helm.go:121]: this looks like duplicated code from https://github.com/gardener/landscaper/blob/7b3f64a3298798f4e92232e6aea5be426f409575/pkg/landscaper/controllers/installations/registry.go#L78 .
I think it makes sense to add it as util function https://github.com/gardener/landscaper/blob/4edfac63889d072622f8951a401c8eee34981aab/pkg/utils/kubernetes/kubernetes.go

**Review by schrodit — CHANGES_REQUESTED** [review]
Thanks for the contribution:)

I jsut have some minor nits the rest looks good

File: pkg/deployer/helm/helm.go (hunk 1 above)
Comment (loewenstein) [reply pkg/deployer/helm/helm.go]: Done

File: pkg/deployer/helm/helm.go (hunk 2 above)
Comment (loewenstein) [reply pkg/deployer/helm/helm.go]: Done

File: pkg/utils/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -79,6 +79,19 @@ func CreateOrUpdate(ctx context.Context, c client.Client, obj client.Object, f c
 	return controllerutil.CreateOrUpdate(ctx, c, obj, f)
 }
 
+func ResolveSecrets(ctx context.Context, client client.Client, secretRefs []v1alpha1.ObjectReference) ([]corev1.Secret, error) {
```
Comment (schrodit) [inline pkg/utils/kubernetes/kubernetes.go:82]: can you add a description about what the function is doing.

**Review by schrodit — CHANGES_REQUESTED** [review]

File: pkg/utils/kubernetes/kubernetes.go (hunk 3 above)
Comment (loewenstein) [reply pkg/utils/kubernetes/kubernetes.go:83]: Done

File: pkg/utils/kubernetes/kubernetes.go [hunk 4]
```diff
@@ -79,6 +79,20 @@ func CreateOrUpdate(ctx context.Context, c client.Client, obj client.Object, f c
 	return controllerutil.CreateOrUpdate(ctx, c, obj, f)
 }
 
+// Find and return the secrets referenced by secretRefs
```
Comment (schrodit) [inline pkg/utils/kubernetes/kubernetes.go:82]:
```suggestion
// ResolveSecrets finds and returns the secrets referenced by secretRefs
```
see the go doc best practices: https://blog.golang.org/godoc

**Review by schrodit — CHANGES_REQUESTED** [review]

File: pkg/utils/kubernetes/kubernetes.go (hunk 4 above)
Comment (loewenstein) [reply pkg/utils/kubernetes/kubernetes.go]: Done (by amending to the original commit).
How relevant is 
> Amend the affected commit(s) and force push onto your branch.
from the [Gardener Pull Request checklis](https://github.com/gardener/documentation/blob/master/CONTRIBUTING.md#Pull-Request-Checklist)?

Comment (schrodit) [reply pkg/utils/kubernetes/kubernetes.go]: i will anyway squash the commits, so thats fine

**Review by schrodit — APPROVED** [review]
/lgtm

**Pull request merged**